Core/src/org/sleuthkit/autopsy/datamodel/ContentUtils.java

```diff
@@ -404,20 +404,22 @@ public Void visitDir(AbstractFile dir) {
                 int numProcessed = 0;
                 // recurse on children
                 for (Content child : dir.getChildren()) {
-                    java.io.File childFile = getFsContentDest(child);
-                    ExtractFscContentVisitor<T, V> childVisitor
-                            = new ExtractFscContentVisitor<>(childFile, progress, worker, false);
-                    // If this is the source directory of an extract it
-                    // will have a progress and worker, and will keep track
-                    // of the progress bar's progress
-                    if (worker != null && worker.isCancelled()) {
-                        break;
+                    if (child instanceof AbstractFile){
+                        java.io.File childFile = getFsContentDest(child);
+                        ExtractFscContentVisitor<T, V> childVisitor
+                                = new ExtractFscContentVisitor<>(childFile, progress, worker, false);
+                        // If this is the source directory of an extract it
```

Comment:
There are AbstractFiles that are not FsContent objects - LayoutFiles, LocalFiles, DerivedFiles, and SpecialDirectories (VirtualDirectory, LocalDirectory). The code here seems to be focused on FsContent. Is the AbstractFile type check correct here?

Reply:
With it now possible for Directory.getChildren to return an artifact where as it previously only returned AbstractFiles it seemed like checking for instanceof AbstractFile was would maintain the current functionality.  However the existing and correct functionality are not necessarily the same thing, and checking for FsContent does seem like the the correct check given the context.